### PR TITLE
Ship basic apparmor profiles

### PR DESCRIPTION
**Pull request by agx**

### CODE REVIEW

**Comment by fbartels** [comment]
Hello @agx,

thanks for your pull request (and attending our conference last week). I have created https://jira.kopano.io/browse/KC-830 for our internal followup and already opened a pull request with your changes against our working branch on Bitbucket.

Will update this PR (and your other PRs) once we have settled some process discussions (as outlined in the contributors.md we otherwise prefer patches to contributing at kopano dot io).

**Comment by jengelh** [comment]
You unlock `/var/lib/kopano/dagent/plugins/`, but that does not appear to take into account that such python script(s), with intent, could open other files.

**Comment by longsleep** [comment]
> You unlock /var/lib/kopano/dagent/plugins/, but that does not appear to take into account that such python script(s), with intent, could open other files.

This is intentional is it not? For some known and trusted plugins there could be pre-defined rules. For all the rest its fine to deny them imho.

**Comment by jengelh** [comment]
Even if the script does not touch any files, the Python interpreter probably wants to go searching /usr/lib/python and whatnot.

**Comment by agx** [comment]
Hi Felix,
On Mon, Oct 02, 2017 at 08:29:41AM +0000, Felix Bartels wrote:
> Hello @agx,
> 
> thanks for your pull request (and attending our conference last
> week). I have created https://jira.kopano.io/browse/KC-830 for our
> internal followup and already opened a pull request with your changes
> against our working branch on Bitbucket.

Thanks! How would I see the patch attached to that issue?

> 
> Will update this PR (and your other PRs) once we have settled some
> process discussions (as outlined in the contributors.md we otherwise
> prefer patches to contributing at kopano dot io).

I have considered this but the problem with kopano.io is that it doesn't
allow to create accounts (lat alone reuse ones from e.g. github like
e.g. theforeman.org does). And sending patches to non-public lists has
the problem that you don't get a reference back that you can link to.

Cheers,
 -- Guido

**Comment by agx** [comment]
Hi,
On Mon, Oct 02, 2017 at 09:43:23AM +0000, jengelh wrote:
> You unlock `/var/lib/kopano/dagent/plugins/`, but that does not appear
> to take into account that such python script(s), with intent, could
> open other files.

I did not come across these yet, that's why. Since it is in /usr/ I
don't expect the admin to put things there (in contrast to
kopano-server's userscripts) so we should add subprofiles for each
plugin that shows up.

That said: I wouldn't bother that much about the details of _each_
profile yet but rather get the automake changes in to install these in
examples. We can easily patch up things from there and our tests will
show denials and such from there on.

File: installer/linux/Makefile.am [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # -*- Makefile -*-
 
+NULL=
```
Comment (jengelh) [inline installer/linux/Makefile.am]: Mh don't really need that one.

**Comment by agx** [comment]
Hi,
On Mon, Oct 02, 2017 at 10:57:54AM -0700, jengelh wrote:
> jengelh commented on this pull request.
> 
> 
> 
> > @@ -1,5 +1,6 @@
>  # -*- Makefile -*-
>  
> +NULL=
> 
> Mh don't really need that one.

You can leave it out (I often do so too) but it's nicer to make sure NULL
isn't define to s.th. else when you end lists but I can drop it.

**Comment by jengelh** [comment]
I'd also leave out `$(NULL)` altogether.

**Comment by agx** [comment]
Hi,
On Mon, Oct 02, 2017 at 06:05:10PM +0000, jengelh wrote:
> I'd also leave out `$(NULL)` altogether.

Without it you get very ugly diff noise when adding new libs. See e.g.

    https://github.com/libvirt/libvirt/blob/master/src/Makefile.am#L705

It really pays of in the long run.

**Comment by fbartels** [comment]
> How would I see the patch attached to that issue?

Unfortunately with our current setup externals are not able to view open pull requests and you will only be able to see the end result as soon as it has been merged.

> And sending patches to non-public lists has the problem that you don't get a reference back that you can link to.

I understand that problem. But fortunately this is not a mailling list, but a mailbox with dedicated personnel. If you mail to it you'll get a Jira ticket back that you can reference to (same as happened with your ticket here).

**Comment by fbartels** [comment]
Merged in https://stash.kopano.io/projects/KC/repos/kopanocore/commits/41ff798d6e2aa9cc1ee011386ae1d6a069fcdbfc

**Pull request closed without merging**